### PR TITLE
add design doc about restore processes

### PR DESCRIPTION
**Pull request by llamerada-jp**

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -111,3 +144,39 @@ spec:
   expire: 2w # when the MantleBackups generated by this MantleBackupConfig should expire.
   suspend: false # whether the periodic backup is active or not.
 ```
+
+### Restore flow
+
+Precondition: Process will not start until conditions are met.
+- The target MantleBackup must exist and be ready to use.
+
+1. Users crate a `MantleRestore` resource.
```
Comment (satoru-takeuchi) [inline docs/design.md:153]:
```suggestion
1. Users create a `MantleRestore` resource.
```

File: docs/design.md [hunk 2]
```diff
@@ -111,3 +144,39 @@ spec:
...
+2. The controller gets the target MantleBackup from the `MantleRestore` resource.
+3. The controller stores the pool name for the `status.pool` field and cluster ID for the `status.clusterID` field. This value is used to remove the restored PV/PVC when the MantleRestore resource is deleted.
+4. The controller gets backup target RBD snap image name from the MantleBackup.
+5. The controller creates a new RBD clone from the RBD snap image.
```
Comment (satoru-takeuchi) [inline docs/design.md:157]:
```suggestion
5. The controller creates a new RBD clone image from the RBD snapshot.
```

File: docs/design.md [hunk 3]
```diff
@@ -111,3 +144,39 @@ spec:
...
+
+### Restore flow
+
+Precondition: Process will not start until conditions are met.
+- The target MantleBackup must exist and be ready to use.
+
+1. Users crate a `MantleRestore` resource.
+2. The controller gets the target MantleBackup from the `MantleRestore` resource.
+3. The controller stores the pool name for the `status.pool` field and cluster ID for the `status.clusterID` field. This value is used to remove the restored PV/PVC when the MantleRestore resource is deleted.
+4. The controller gets backup target RBD snap image name from the MantleBackup.
```
Comment (satoru-takeuchi) [inline docs/design.md:156]:
```suggestion
4. The controller gets backup target RBD snapshot name from the MantleBackup.
```

File: docs/design.md [hunk 4]
```diff
@@ -111,3 +144,39 @@ spec:
...
+5. The controller creates a new RBD clone from the RBD snap image.
+6. The controller creates a new PV/PVC using the new RBD clone.
```
Comment (satoru-takeuchi) [inline docs/design.md:158]:
```suggestion
6. The controller creates a new PV/PVC using the above-mentioned RBD clone image.
```

File: docs/design.md [hunk 5]
```diff
@@ -111,3 +144,39 @@ spec:
...
+
+### Cleanup restore flow
+
+1. Users delete the `MantleRestore` resource.
+2. The controller tries to delete the PV/PVC created by the `MantleRestore` resource and wait until the PV/PVC is deleted. If the PV/PVC is used by some pod.
```
Comment (satoru-takeuchi) [inline docs/design.md:163]:
```suggestion
2. The controller tries to delete the PV/PVC created by the `MantleRestore` resource and wait until the PV/PVC are deleted. If the PV/PVC are consumed by some pods.
```

File: docs/design.md [hunk 6]
```diff
@@ -111,3 +144,39 @@ spec:
...
+3. The controller removes the RBD clone created by the `MantleRestore` resource. The controller should not remove RBD clone volume specified by the PV.
```
Comment (satoru-takeuchi) [inline docs/design.md:164]: > The controller should not remove RBD clone volume specified by the PV.

Could you tell me the difference between `RBD clone` and `RBD clone volume`? Are they mean the same thing? If so, the first sentence describes the controller removes the RBD clone image and the second sentence describes the controllere should not do so.

**Review by satoru-takeuchi — CHANGES_REQUESTED** [review]

File: docs/design.md (hunk 6 above)
Comment (llamerada-jp) [reply docs/design.md]: The original sentence was misleading. I updated.

**Review by satoru-takeuchi — APPROVED** [review]

**Pull request merged**